crates/wasmparser/src/validator/operators.rs

```diff
@@ -475,7 +478,7 @@ impl<'resources, R: WasmModuleResources> OperatorValidatorTemp<'_, 'resources, R
             if self.operands.len() == control.height {
                 let desc = match expected {
                     Some(ty) => ty_to_str(ty),
-                    None => "a type",
+                    None => "a type".into(),
```

Comment:
Ah I think this is where my perf-sensitive thinking is coming from. Given that failure is the slow path though I think it's reasonable to outline this and mark it `#[cold]`. (similar to `self.err_beyond_end` IIRC)

If you wouldn't mind double-checking that this is the only place though that'd be good I think